master/src/main/scala/org/apache/celeborn/service/deploy/master/quota/QuotaManager.scala

```diff
@@ -37,4 +38,74 @@ class QuotaManager(celebornConf: CelebornConf, configService: ConfigService) ext
       DEFAULT_QUOTA
     }
   }
+
+  def checkQuotaSpaceAvailable(
+      userIdentifier: UserIdentifier,
+      resourceResumption: ResourceConsumption): (Boolean, String) = {
+    val quota = getQuota(userIdentifier)
+    val checkResults = Seq(
+      checkDiskBytesWritten(userIdentifier, resourceResumption.diskBytesWritten, quota),
+      checkDiskFileCount(userIdentifier, resourceResumption.diskFileCount, quota),
+      checkHdfsBytesWritten(userIdentifier, resourceResumption.hdfsBytesWritten, quota),
+      checkHdfsFileCount(userIdentifier, resourceResumption.hdfsFileCount, quota))
+    val exceed = checkResults.foldLeft(false)(_ || _._1)
+    val reason = checkResults.foldLeft("")(_ + _._2)
+    (!exceed, reason)
+  }
+
+  private def checkDiskBytesWritten(
```

Comment:
Could the check methods merge into one method? Most implementation of check methods are similar.

Reply:
> Could the check methods merge into one method? Most implementation of check methods are similar.

Let me do a final refactor pr for all code related to quota management.